### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by martinkarugaba**
# :recycle:  Changes made

- [ ] Solved the N+1 problem when fetching all posts and their comments for a user.

## Before solving N+1 problem
before

## After solving N+1 problem
after

## 1. user index page:

- [ ] I can see the username of all other users.
- [ ] I can see the profile picture for each user.
- [ ] I can see the number of posts each user has written.
- [ ] When I click on a user, I am redirected to that user's show page.

## 2. For user show page:

- [ ] I can see the user's profile picture.
- [ ] I can see the user's username.
- [ ] I can see the number of posts the user has written.
- [ ] I can see the user's bio.
- [ ] I can see the user's first 3 posts.
- [ ] I can see a button that lets me view all of a user's posts.
- [ ] When I click a user's post, it redirects me to that post's show page.
- [ ] When I click to see all posts, it redirects me to the user's post's index page.

## 3. For the User post index page:

- [ ] I can see the user's profile picture.
- [ ] I can see the user's username.
- [ ] I can see the number of posts the user has written.
- [ ] I can see a post's title.
- [ ] I can see some of the post's body.
- [ ] I can see the first comments on a post.
- [ ] I can see how many comments a post has.
- [ ] I can see how many likes a post has.
- [ ] I can see a section for pagination if there are more posts than fit on the view.
- [ ] When I click on a post, it redirects me to that post's show page.

## 4. Post show page:

- [ ] I can see the post's title.
- [ ] I can see who wrote the post.
- [ ] I can see how many comments it has.
- [ ] I can see how many likes it has.
- [ ] I can see the post body.
- [ ] I can see the username of each commentor.
- [ ] I can see the comment each commentor left.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
     @page = params.fetch(:page, 0).to_i
-    @posts = @user.posts.limit(3).offset(@page * 3)
+    @posts = @user.posts.includes(:author).limit(3).offset(@page * 3)
```
Comment (TedLivist) [inline app/controllers/posts_controller.rb]: - [ ] You should be including `:comments`, not `:author`. The `@user` is already the `author`. The point eliminating N+1 queries is to limit the number of round trips to fetch the comments that belongs to the posts.

  ```
  @user.posts.includes(:comments)...
  ```

File: README.md [hunk 2]
```diff
@@ -90,11 +90,18 @@ Run `bundle exec rspec` to run tests
 - Twitter: [@martin_karugaba](https://twitter.com/martin_karugaba)
 - LinkedIn: [martinkarugaba](https://www.linkedin.com/in/martinkarugaba/)
 
+### Richard Sikaonga
+
```
Comment (TedLivist) [inline README.md:94]: - [ ] In your PR description you should include the number before and after screenshot of the eliminating the N+1 query problem

File: spec/features/post_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    post = @posts.first
+    visit user_post_path(@user, post)
+    expect(page).to have_content("Comments: #{post.comments.count}")
+  end
+
+  it 'I can see how many likes a post has' do
+    post = @posts.first
+    visit user_post_path(@user, post)
+    expect(page).to have_content(post.likes.count.to_s)
+  end
```
Comment (TedLivist) [inline spec/features/post_index_spec.rb]: - [ ] In these test blocks you are visiting the post show page, not the required post index page for this test file. Kindly visit the correct routes and ensure no test specification is missing.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @martinkarugaba and @richie1988,

Good job so far!
✔️ Well-written and passing tests
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @martinkarugaba, and @richie1988👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- Good job fixing n+1 problem. ✔️
- Good job with integration specs. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️

- [ ] All of the tests are failing. Please make sure to fix the issues and all tests should pass without any errors. (If tests are passing on your end, Please add a screenshot of passed tests in the comment).
image

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by martinkarugaba** [comment]
# All the tests are passing on my end.

- [ ] Below is a screenshot just as requested
 
passing_tests

**Review by zeff96 — APPROVED** [review]
### Status: APPROVED :green_circle: 
Hi @martinkarugaba and @richie1988 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

congrats

## Highlights
:heavy_check_mark: Linters passing
:heavy_check_mark: All test passing
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**